### ISSUE TITLE
Dimension controls: Order by horizontal & vertical

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#51680]
## What problem does this address?
The new dimension controls are awesome!
However, there is a UX inconsistency in the order of the controls when switching from the default to "Custom".
The default is: Top / Bottom and Left / Right (= Horizontal & vertical)
The custom order is: Top, Right, Bottom, Left. (= Mixed)

| GB 16.0 Default Horizontal & vertical |
|--------|
| <img width="282" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/61202535-80c7-453d-9526-b2c3c6de42f6">
 | 

## What is your proposed solution?
The custom order should reflect what's already displayed on the default, meaning:
- Top
- Bottom
- Left
- Right

Maybe we can even add a tiny bit of padding between "bottom" and "left" subtly distinguish between Horizontal & Vertical
Here is a quick mockup:
| GB 16.0 Custom (Mixed) | Proposal (Keeping Horizontal & Vertical order) |
|--------|--------|
| <img width="280" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/f7abdfb4-077f-4f00-86f8-3fba506e5f5e"> | <img width="280" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/4496845d-e11d-47fd-b8a3-199ceeb39a32"> |

**Comment by hanneslsm** [comment]
The new order should of course also be adapted in the Menu then
<img width="273" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/8acae857-9dcf-445b-b325-01c0db7a66ae">

**Comment by richtabor** [comment]
It's tricky, as top right bottom left is the order in which the CSS shorthand property is written ([source](https://developer.mozilla.org/en-US/docs/Web/CSS/padding)).

**Comment by hanneslsm** [comment]
> It's tricky, as top right bottom left is the order in which the CSS shorthand property is written ([source](https://developer.mozilla.org/en-US/docs/Web/CSS/padding)).

Yeah, I know, but that's not an argument I can bring up with clients. They surely will get confused why the order has changed. The user journey (getting primed with an order already) and target group (not technical users) are different for Gutenberg than for CSS, so I'd argue that we finally break away from the technical constrains and build a more intuitive interface.

Also, one could argue that those ("advanced" padding and margin) features are mainly for (web-)designers. In that case, let's see what Figma does with borders (they even have a different name for it: "strokes"):
<img width="122" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/2c566699-8cea-42db-a1b1-44251c91c739">
